homeassistant/components/squeezebox/entity.py

```diff
@@ -0,0 +1,30 @@
+"""Base class for Squeezebox Sensor entities."""
+
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity import EntityDescription
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN, STATUS_QUERY_UUID
+from .coordinator import LMSStatusDataUpdateCoordinator
+
+
+class LMSStatusEntity(CoordinatorEntity[LMSStatusDataUpdateCoordinator]):
+    """Defines a base status sensor entity."""
+
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        coordinator: LMSStatusDataUpdateCoordinator,
+        description: EntityDescription,
+    ) -> None:
+        """Initialize status sensor entity."""
+        super().__init__(coordinator, context=description.key)
+        self.coordinator = coordinator
+        self.entity_description = description
+        self._attr_name = description.key
+        self._attr_unique_id = f"{coordinator.data[STATUS_QUERY_UUID]}{description.key}"
+
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, coordinator.data[STATUS_QUERY_UUID])},
+        )
```

Comment:
Why don't we create the device here?

Reply:
Means taking stuff into the entity and it was how airgradient did it ;-)

Reply:
Oh, but that's because we don't have all the info in all the entities because we have 2 coordinators. But I think you can just complete the device info here

Reply:
Yes think I might well end up there too, with more entity types, but having thought about It quite a lot last night this seem to be the cleanest approach as it contains all the service device setup stuff to __init__ and then I can just link to that device.